pkgs/development/ocaml-modules/raven/rune.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  buildDunePackage,
+  alcotest,
+  ctypes,
+  ctypes-foreign,
+  dune-configurator,
+  integers,
+  libllvm,
+  raven,
+  raven-nx,
+}:
+
+buildDunePackage {
+  pname = "rune";
+
+  inherit (raven) version src postUnpack;
+
+  sandboxProfile = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    (allow iokit-open)
+    (allow file-read* (subpath "/System/Library/Extensions"))
+    (allow mach-lookup (global-name "com.apple.MTLCompilerService"))
+  '';
```

Comment:
I am not familiar with this (and macOS in general). Can you explain in a few words why this is required and what this is doing?